[VUMM-297] Move artifact store to common module

anandjakhaniya opened the pull request:
<!-- Title of the PR must comply with [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) guidelines. --> 
<!-- Title should include the JIRA ticket in square brackets after the conventional commit prefix. This will automatically link the PR in JIRA. -->
<!-- Example: "fix: [JIRA-123] Allow creation of groups with no members" -->

## Impact and Context
<!-- Brief description of why these changes are necessary. Don't rewrite the entire Jira ticket. -->
We have an artifact store with two different options which are S3 and NFS. so for that, we have created two services using spring boot to serve artifact store functionality. In the current situation, This functionality is used by two backends which are Registry and ModelDB and both the backends have their own code so whenever we have updated something in one place we have to manage it on the other backends as well. so to protect duplication and uniqueness of the code we need these changes in a common place so that both backends are used same code and we can manage and maintain the code from one place.

There was an old PR: https://github.com/VertaAI/modeldb/pull/2476 but it is too old and takes more time to resolve merge conflicts and make it work so I have created this PR from the latest code with some better code improvement.

## Risks and Area of Effect
<!--
  Describe both risk (how likely this is to break) and area of effect (how wide potential breakages could reach).
  These should be smaller scale than those documented in a design doc.
-->

## Testing
<!-- Explain how this contribution has been tested, e.g. what tests were added and where they have been run. -->

## How to Revert
<!-- List steps required to revert this change. For example, note if we'd need to revert liquibase changes. -->

anandjakhaniya committed:
Fix MDB initialization using spring boot

Commit:
https://jenkins.dev.verta.ai/job/build/job/autoformat/job/modeldb-backend/881/

anandjakhaniya committed:
Fix code

Commit:
https://jenkins.dev.verta.ai/job/build/job/autoformat/job/modeldb-backend/883/

anandjakhaniya committed:
Fix pom

anandjakhaniya committed:
Merge branch 'aj/fix-initialization' of github.com:VertaAI/modeldb into aj/fix-initialization

anandjakhaniya committed:
improve code

Commit:
https://jenkins.dev.verta.ai/job/build/job/autoformat/job/modeldb-backend/890/

anandjakhaniya committed:
improve code

Commit:
https://jenkins.dev.verta.ai/job/build/job/autoformat/job/modeldb-backend/895/

anandjakhaniya committed:
Improve code

anandjakhaniya committed:
Merge branch 'aj/fix-initialization' of github.com:VertaAI/modeldb into aj/fix-initialization

Commit:
https://jenkins.dev.verta.ai/job/build/job/autoformat/job/modeldb-backend/897/

anandjakhaniya committed:
[VUMM-297] Move artifact store to common module

Commit:
Update protos from aj/move-artifact-store-common

anandjakhaniya committed:
Fix compilation for proto changes

anandjakhaniya committed:
Fix test compilation

Commit:
Update protos from https://github.com/VertaAI/protos-all/pull/336

Commit:
Update protos from https://github.com/VertaAI/protos-all/pull/336

Commit:
Update protos from https://github.com/VertaAI/protos-all/pull/336

anandjakhaniya committed:
revert client breaking changes

anandjakhaniya committed:
Fix test init for artifact store

Commit:
https://jenkins.dev.verta.ai/job/build/job/autoformat/job/modeldb-backend/914/

anandjakhaniya commented:
!rebuild

anandjakhaniya committed:
improve code

Commit:
https://jenkins.dev.verta.ai/job/build/job/autoformat/job/modeldb-backend/917/

anandjakhaniya committed:
Merge branch 'main' into aj/fix-initialization

# Conflicts:
#	backend/src/main/java/ai/verta/modeldb/App.java
#	backend/src/main/java/ai/verta/modeldb/ServiceSet.java

anandjakhaniya committed:
Move port collision logic to common app class

anandjakhaniya committed:
improve code

anandjakhaniya committed:
Merge branch 'aj/fix-initialization' into aj/move-artifact-store-common

# Conflicts:
#	backend/common/src/main/java/ai/verta/modeldb/common/CommonApp.java
#	backend/common/src/main/java/ai/verta/modeldb/common/CommonConstants.java
#	backend/src/main/java/ai/verta/modeldb/App.java

Commit:
https://jenkins.dev.verta.ai/job/build/job/autoformat/job/modeldb-backend/922/

anandjakhaniya committed:
Merge branch 'main' into aj/fix-initialization

anandjakhaniya committed:
Merge branch 'main' into aj/fix-initialization

# Conflicts:
#	backend/common/src/main/java/ai/verta/modeldb/common/config/Config.java
#	backend/src/main/java/ai/verta/modeldb/App.java

anandjakhaniya committed:
Move OpenTelemetry to common app class

anandjakhaniya committed:
reverted last commit [Move OpenTelemetry to common app class]

jkwatson-verta commented on backend/common/src/main/java/ai/verta/modeldb/common/CommonConstants.java:
```diff
@@ -17,6 +17,12 @@ public interface CommonConstants {
   String LIQUIBASE_MIGRATION = "LIQUIBASE_MIGRATION";
   String RUN_LIQUIBASE_SEPARATE = "RUN_LIQUIBASE_SEPARATE";
   String BACKEND_PID = "verta-backend.pid";
+  String FILENAME = "FileName";
```
this looks like it's being used for a very specific purpose, i.e. a query parameter. Recommend suffixing it with something that identifies what it's being used for (like "FILENAME_QUERY_PARAMETER"). Also, is this "common" enough to be in this pool of constants that all of our java services will be accessing?

anandjakhaniya replied:
`String FILENAME = "FileName";` is used for multiple purpose like you said as query parameter also used in artifact store endpoint path url. So i have make it just FILENAME instead of duplicating with two like
```
String FILENAME_QUERY_PARAMETER = "FileName"
String FILENAME_ARTIFACT_FILE_PATH_PREFIX = "FileName"
```

anandjakhaniya replied:
> is this "common" enough to be in this pool of constants that all of our java services will be accessing?

Yes it is used across multiple backends.

jkwatson-verta replied:
I would prefer to have two different constants if they are being used for two different purposes. Otherwise, it's too easy to break one when you change for the other.

anandjakhaniya replied:
Fixed

jkwatson-verta commented on backend/common/src/main/java/ai/verta/modeldb/common/CommonConstants.java:
```diff
@@ -17,6 +17,12 @@ public interface CommonConstants {
   String BACKEND_PID = "verta-backend.pid";
+  String FILENAME = "FileName";
+  String PUT = "put";
+  String GET = "get";
```
very strongly recommend against putting simple words like this into a constants file. Just use them as-is, in-line.

anandjakhaniya replied:
this is used across multiple backend so added it in CommonConstants.java
It is used multiple places so as per the sonarqube suggestions i have extracted it in constant file.

jkwatson-verta replied:
we should adjust our sonarqube settings to not flag this, then. Putting any arbitrary duplicated constant in a separate file is silly and just makes the code more confusing.

anandjakhaniya replied:
Fixed

anandjakhaniya committed:
Merge branch 'aj/fix-initialization' into aj/move-artifact-store-common

# Conflicts:
#	backend/common/src/main/java/ai/verta/modeldb/common/config/Config.java
#	backend/src/main/java/ai/verta/modeldb/App.java

anandjakhaniya committed:
Merge branch 'main' into aj/move-artifact-store-common

# Conflicts:
#	backend/common/pom.xml
#	backend/common/src/main/java/ai/verta/modeldb/common/CommonConstants.java
#	backend/common/src/main/java/ai/verta/modeldb/common/artifactStore/storageservice/s3/RefCountedS3Client.java
#	backend/common/src/main/java/ai/verta/modeldb/common/config/Config.java
#	backend/src/main/java/ai/verta/modeldb/App.java
#	backend/src/main/java/ai/verta/modeldb/ServiceSet.java
#	backend/src/main/java/ai/verta/modeldb/experimentRun/FutureExperimentRunDAO.java
#	backend/src/main/java/ai/verta/modeldb/health/HealthServiceImpl.java
#	backend/src/main/java/ai/verta/modeldb/project/FutureProjectDAO.java
#	backend/src/test/java/ai/verta/modeldb/TestsInit.java

Commit:
https://jenkins.dev.verta.ai/job/build/job/autoformat/job/modeldb-backend/1096/

anandjakhaniya committed:
Fix code and remove unrelated changes

anandjakhaniya committed:
Merge branch 'aj/move-artifact-store-common' of github.com:VertaAI/modeldb into aj/move-artifact-store-common

anandjakhaniya committed:
Fix code

Commit:
https://jenkins.dev.verta.ai/job/build/job/autoformat/job/modeldb-backend/1099/

jkwatson-verta commented on backend/common/src/main/java/ai/verta/modeldb/common/artifactStore/storageservice/ArtifactStoreService.java:
```diff
@@ -100,7 +100,7 @@ default String getUrl(
     for (Map.Entry<String, Object> queryParam : parameters.entrySet()) {
       // Adding the filename to the path to enable file saved with that name
-      if (queryParam.getKey().equals(ModelDBConstants.FILENAME)) {
+      if (queryParam.getKey().equals(CommonConstants.FILENAME)) {
```
please rename constant to `FILENAME_QUERY_PARAM`

anandjakhaniya replied:
Fixed

jkwatson-verta commented on backend/common/src/main/java/ai/verta/modeldb/common/artifactStore/storageservice/nfs/NFSController.java:
```diff
@@ -78,7 +78,7 @@ public ResponseEntity<Resource> getArtifact(
               HttpHeaders.CONTENT_DISPOSITION,
               "attachment; filename=\"" + resource.getFilename() + "\"")
-          .header(ModelDBConstants.FILENAME, fileName)
+          .header(CommonConstants.FILENAME, fileName)
```
rename here to `FILENAME_HTTP_HEADER`

anandjakhaniya replied:
Fixed

jkwatson-verta commented on backend/common/src/main/java/ai/verta/modeldb/common/artifactStore/storageservice/nfs/NFSService.java:
```diff
@@ -157,24 +155,24 @@ private String generatePresignedUrl(String artifactPath, String method) {
       LOGGER.trace("NFSService - generatePresignedUrl - get url returned");
       var filename = artifactPath.substring(artifactPath.lastIndexOf("/"));
-      parameters.put(ModelDBConstants.FILENAME, filename);
+      parameters.put(CommonConstants.FILENAME, filename);
```
`FILENAME_QUERY_PARAMETER`

anandjakhaniya replied:
Fixed

jkwatson-verta commented on backend/common/src/main/java/ai/verta/modeldb/common/artifactStore/storageservice/s3/S3Controller.java:
```diff
@@ -51,7 +51,7 @@ public ResponseEntity<UploadFileResponse> storeArtifact(
   @GetMapping(value = {"${artifactEndpoint.getArtifact}/{FileName}"})
   public ResponseEntity<Resource> getArtifact(
-      @PathVariable(value = ModelDBConstants.FILENAME) String fileName,
+      @PathVariable(value = CommonConstants.FILENAME_QUERY_PARAMETER) String fileName,
```
this isn't a query parameter here, is it? It's a path parameter? I think this making "Filename" a "common" constant is a mistake. It's used in too many ways, and changing one of them would inadvertently break the others.

anandjakhaniya replied:
Fixed

jkwatson-verta commented on backend/common/src/main/java/ai/verta/modeldb/common/artifactStore/storageservice/s3/S3Service.java:
```diff
@@ -279,31 +278,31 @@ public String getPresignedUrlViaMDB(
       LOGGER.debug("S3Service - generatePresignedUrl - returning " + method + " url");
       var filename = artifactPath.substring(artifactPath.lastIndexOf("/"));
-      parameters.put(ModelDBConstants.FILENAME, filename);
+      parameters.put(CommonConstants.FILENAME, filename);
```
query parameter here, yes?

anandjakhaniya replied:
Removed from common and use static and fixed

jkwatson-verta commented on backend/src/main/java/ai/verta/modeldb/experiment/FutureExperimentDAO.java:
```diff
@@ -875,7 +875,24 @@ public InternalFuture<GetUrlForArtifact.Response> getUrlForArtifact(GetUrlForArt
-    return permissionCheck.thenCompose(
-        unused -> artifactHandler.getUrlForArtifact(request), executor);
+    return permissionCheck
+        .thenCompose(
+            unused ->
+                artifactHandler.getUrlForArtifact(
+                    ai.verta.common.GetUrlForArtifact.newBuilder()
+                        .setId(request.getId())
+                        .setKey(request.getKey())
+                        .setMethod(request.getMethod())
+                        .setPartNumber(request.getPartNumber())
+                        .setArtifactType(request.getArtifactType())
+                        .build()),
+            executor)
+        .thenApply(
+            response ->
+                GetUrlForArtifact.Response.newBuilder()
+                    .setUrl(response.getUrl())
+                    .setMultipartUploadOk(response.getMultipartUploadOk())
+                    .build(),
+            executor);
```
this looks like a functional change. I'm not sure how this in part of making a common module.

anandjakhaniya replied:
this change is part of common module and we have added protos changes here: https://github.com/VertaAI/protos-all/pull/336 to make Artifact store code in common module and then we can use it in java like above.

jkwatson-verta replied:
do we have tests for this?

jkwatson-verta commented on backend/src/main/java/ai/verta/modeldb/experimentRun/FutureExperimentRunDAO.java:
```diff
@@ -705,8 +706,25 @@ public InternalFuture<GetUrlForArtifact.Response> getUrlForArtifact(GetUrlForArt
-    return permissionCheck.thenCompose(
-        unused -> artifactHandler.getUrlForArtifact(request), executor);
+    return permissionCheck
+        .thenCompose(
+            unused ->
+                artifactHandler.getUrlForArtifact(
+                    ai.verta.common.GetUrlForArtifact.newBuilder()
+                        .setId(request.getId())
+                        .setKey(request.getKey())
+                        .setMethod(request.getMethod())
+                        .setPartNumber(request.getPartNumber())
+                        .setArtifactType(request.getArtifactType())
+                        .build()),
+            executor)
+        .thenApply(
+            response ->
+                GetUrlForArtifact.Response.newBuilder()
+                    .setUrl(response.getUrl())
+                    .setMultipartUploadOk(response.getMultipartUploadOk())
+                    .build(),
+            executor);
```
this also looks like a functional change.

anandjakhaniya replied:
this change is part of common module and we have added protos changes here: https://github.com/VertaAI/protos-all/pull/336 to make Artifact store code in common module and then we can use it in java like above

jkwatson-verta replied:
tests needed

jkwatson-verta commented on backend/src/main/java/ai/verta/modeldb/versioning/BlobDAORdbImpl.java:
```diff
@@ -1728,7 +1730,7 @@ public GetUrlForDatasetBlobVersioned.Response getUrlForVersionedDatasetBlob(
                     datasetId,
                     request.getDatasetVersionId(),
-                    request.getMethod().equalsIgnoreCase(ModelDBConstants.PUT)),
+                    request.getMethod().equalsIgnoreCase(CommonConstants.PUT)),
```
i thought we were getting rid of these http verb constants.

anandjakhaniya replied:
Fixed

jkwatson-verta commented on backend/common/src/main/java/ai/verta/modeldb/common/config/ArtifactStoreConfig.java:
```diff
@@ -1,17 +1,47 @@
+
+    switch (getArtifactStoreType()) {
+      case "S3":
+        if (S3 == null) throw new InvalidConfigException(base + ".S3", Config.MISSING_REQUIRED);
```
we should always put brackets around our `if` blocks.

liam-verta replied:
And we should enforce this.  Via Sonarqube?

anandjakhaniya replied:
Fixed

jkwatson-verta commented on backend/common/src/main/java/ai/verta/modeldb/common/config/ArtifactStoreConfig.java:
```diff
@@ -1,17 +1,47 @@
+        break;
+      case "NFS":
+        if (getNFS() == null)
+          throw new InvalidConfigException(base + ".NFS", Config.MISSING_REQUIRED);
```
brackets, please

anandjakhaniya replied:
Fixed

jkwatson-verta commented on backend/common/src/main/java/ai/verta/modeldb/common/config/ArtifactStoreConfig.java:
```diff
@@ -1,17 +1,47 @@
   private String host = "";
 
+  @SuppressWarnings({"squid:S116"})
+  private S3Config S3;
```
fields should always be lower case

anandjakhaniya replied:
Fixed

jkwatson-verta commented on backend/common/src/main/java/ai/verta/modeldb/common/config/ArtifactStoreConfig.java:
```diff
@@ -1,17 +1,47 @@
 
   private NFSEndpointConfig artifactEndpoint;
 
+  public void Validate(String base) throws InvalidConfigException {
```
method should be lower case. Also, do we have tests for this?

anandjakhaniya replied:
Fixed

jkwatson-verta commented on backend/common/src/main/java/ai/verta/modeldb/common/config/ArtifactStoreConfig.java:
```diff
@@ -40,7 +74,25 @@ public NFSEndpointConfig getArtifactEndpoint() {
   }
 
-  public abstract String storeTypePathPrefix();
+  public String storeTypePathPrefix() {
```
tests?

jkwatson-verta commented on backend/common/src/main/java/ai/verta/modeldb/common/config/ArtifactStoreConfig.java:
```diff
@@ -40,7 +74,25 @@ public NFSEndpointConfig getArtifactEndpoint() {
+  }
 
-  public abstract String getPathPrefixWithSeparator();
+  public String getPathPrefixWithSeparator() {
```
tests?

jkwatson-verta commented on backend/common/src/main/java/ai/verta/modeldb/common/config/Config.java:
```diff
@@ -60,6 +60,7 @@ public abstract class Config {
   private TracingServerInterceptor tracingServerInterceptor = null;
   private TracingClientInterceptor tracingClientInterceptor = null;
   private volatile OpenTelemetry openTelemetry;
+  public ArtifactStoreConfig artifactStoreConfig;
```
please keep fields private

anandjakhaniya replied:
Fixed

jkwatson-verta commented on backend/src/main/java/ai/verta/modeldb/experimentRun/FutureExperimentRunDAO.java:
```diff
@@ -715,23 +733,51 @@ public InternalFuture<GetCommittedArtifactParts.Response> getCommittedArtifactPa
+                GetCommittedArtifactParts.Response.newBuilder()
+                    .addAllArtifactParts(response.getArtifactPartsList())
+                    .build(),
+            executor);
```
tests needed

jkwatson-verta commented on backend/src/main/java/ai/verta/modeldb/experimentRun/FutureExperimentRunDAO.java:
```diff
@@ -715,23 +733,51 @@ public InternalFuture<GetCommittedArtifactParts.Response> getCommittedArtifactPa
+                        .setKey(request.getKey())
+                        .setArtifactPart(request.getArtifactPart())
+                        .build()),
+            executor);
```
tests needed

jkwatson-verta commented on backend/src/main/java/ai/verta/modeldb/experimentRun/FutureExperimentRunDAO.java:
```diff
@@ -715,23 +733,51 @@ public InternalFuture<GetCommittedArtifactParts.Response> getCommittedArtifactPa
+                        .setId(request.getId())
+                        .setKey(request.getKey())
+                        .build()),
+            executor);
```
tests

jkwatson-verta commented on backend/src/main/java/ai/verta/modeldb/project/FutureProjectDAO.java:
```diff
@@ -441,8 +441,25 @@ public InternalFuture<GetUrlForArtifact.Response> getUrlForArtifact(GetUrlForArt
+                    .setUrl(response.getUrl())
+                    .setMultipartUploadOk(response.getMultipartUploadOk())
+                    .build(),
+            executor);
```
tests needed for any functional changes here.

anandjakhaniya committed:
Merge branch 'main' into aj/move-artifact-store-common

# Conflicts:
#	backend/common/src/main/java/ai/verta/modeldb/common/CommonConstants.java
#	backend/common/src/main/java/ai/verta/modeldb/common/config/Config.java
#	backend/src/main/java/ai/verta/modeldb/config/MDBArtifactStoreConfig.java
#	backend/src/main/java/ai/verta/modeldb/config/MDBConfig.java
#	backend/src/main/java/ai/verta/modeldb/configuration/AppConfigBeans.java

Commit:
Update protos from https://github.com/VertaAI/protos-all/pull/336

anandjakhaniya committed:
Fix protos and compilation and updated from main

jkwatson-verta commented on backend/common/src/main/java/ai/verta/modeldb/common/CommonMessages.java:
```diff
@@ -1,6 +1,10 @@
 public abstract class CommonMessages {
+
+  public static final String ARTIFACT_STORE_DISABLED_LOGS = "Artifact store is disabled";
+  public static final String BUCKET_DOES_NOT_EXISTS = "Bucket does not exists";
```
```suggestion
  public static final String BUCKET_DOES_NOT_EXIST = "Bucket does not exist";
```

anandjakhaniya committed:
Fix review comments

anandjakhaniya committed:
Fix test compilation

Commit:
https://jenkins.dev.verta.ai/job/build/job/autoformat/job/modeldb-backend/116/

anandjakhaniya commented:
@jkwatson-verta I have already added many test cases but it is not feasible to run it with automation so I have disabled it or either not added in test suits. We should fix that in separate PR with a separate ticket.

Test cases: 
- https://github.com/VertaAI/modeldb/blob/96d104b2bf2737989fe729f7828a08fe6bea2bf0/backend/src/test/java/ai/verta/modeldb/ExperimentRunTest.java#L4542
- https://github.com/VertaAI/modeldb/blob/main/backend/src/test/java/ai/verta/modeldb/ArtifactStoreTest.java
- https://github.com/VertaAI/modeldb/blob/main/backend/src/test/java/ai/verta/modeldb/NFSArtifactStoreTest.java

jkwatson-verta commented on backend/common/src/main/java/ai/verta/modeldb/common/config/Config.java:
```diff
@@ -96,6 +97,10 @@ public void validate() throws InvalidConfigException {
     springServer.validate("springServer");
+
+    if (artifactStoreConfig == null)
+      throw new InvalidConfigException("artifactStoreConfig", MISSING_REQUIRED);
```
needs some brackets

anandjakhaniya replied:
Added

jkwatson-verta commented on backend/src/main/java/ai/verta/modeldb/config/TestConfig.java:
```diff
@@ -38,9 +38,9 @@ public void validate() throws InvalidConfigException {
 
-    if (artifactStoreConfig == null)
+    if (getArtifactStoreConfig() == null)
       throw new InvalidConfigException("artifactStoreConfig", MISSING_REQUIRED);
```
needs brackets

anandjakhaniya replied:
Added

jkwatson-verta commented on backend/src/main/java/ai/verta/modeldb/utils/JdbiUtil.java:
```diff
@@ -0,0 +1,104 @@
+    if (jdbiUtil == null) {
+      jdbiUtil = new JdbiUtil();
+      config = mdbConfig;
+      liquibaseRootFilePath = "\\src\\main\\resources\\liquibase\\db-changelog-master.xml";
```
curious why we're using windows file separators here. We should be able to use unix ones and have java do the right thing, correct?

anandjakhaniya replied:
Java only taking this `"\\src\\main\\resources\\liquibase\\db-changelog-master.xml";` and it will automatically handled with unix and windows both.

anandjakhaniya replied:
If I am setting `"/src/main/resources/liquibase/db-changelog-master.xml";` then the java process will fail and say about fix the path.

jkwatson-verta replied:
that's ugly. so, windows separators work on linux, but not the other way around?

anandjakhaniya replied:
@jkwatson-verta this `"\\src\\main\\resources\\liquibase\\db-changelog-master.xml";` is not specific to windows. the Path is actually `"\src\main\resources\liquibase\db-changelog-master.xml";` but In the java, `\` is pointing special character so we have to add one another `\` before that so the java is suggesting it so I don't think it is only related to windows.

jkwatson-verta replied:
only windows uses the backslash as a path separator. :)

jkwatson-verta commented:
This looks generally ok. If we want to save testing until later, that's ok, but we should create an issue to track it.

anandjakhaniya committed:
Fix review comment about brackets

Commit:
https://jenkins.dev.verta.ai/job/build/job/autoformat/job/modeldb-backend/140/

anandjakhaniya committed:
Merge branch 'main' into aj/move-artifact-store-common

# Conflicts:
#	backend/src/test/java/ai/verta/modeldb/TestsInit.java

jkwatson-verta commented:
This PR is gigantic, and to my eyes, includes functional changes with no tests for them. I don't want to merge this without testing on the new functionality.

anandjakhaniya commented:
> This PR is gigantic, and to my eyes, includes functional changes with no tests for them. I don't want to merge this without testing on the new functionality.

https://github.com/VertaAI/modeldb/pull/3003#issuecomment-1201160949

@jkwatson-verta I have already added many test cases but it is not feasible to run it with automation so I have disabled it or either not added in test suits. We should fix that in separate PR with a separate ticket.

Test cases: 
- https://github.com/VertaAI/modeldb/blob/96d104b2bf2737989fe729f7828a08fe6bea2bf0/backend/src/test/java/ai/verta/modeldb/ExperimentRunTest.java#L4542
- https://github.com/VertaAI/modeldb/blob/main/backend/src/test/java/ai/verta/modeldb/ArtifactStoreTest.java
- https://github.com/VertaAI/modeldb/blob/main/backend/src/test/java/ai/verta/modeldb/NFSArtifactStoreTest.java

jkwatson-verta commented:
> @jkwatson-verta I have already added many test cases but it is not feasible to run it with automation so I have disabled it or either not added in test suits. We should fix that in separate PR with a separate ticket.
> 
> Test cases:
> 
> * https://github.com/VertaAI/modeldb/blob/96d104b2bf2737989fe729f7828a08fe6bea2bf0/backend/src/test/java/ai/verta/modeldb/ExperimentRunTest.java#L4542
> * https://github.com/VertaAI/modeldb/blob/main/backend/src/test/java/ai/verta/modeldb/ArtifactStoreTest.java
> * https://github.com/VertaAI/modeldb/blob/main/backend/src/test/java/ai/verta/modeldb/NFSArtifactStoreTest.java

ok. we need to get modeldb tests running with automation as soon as possible, in my opinion. @eheinlein-verta can we prioritize this work?

jkwatson-verta commented:
Since the Jira ticket has no reason as to why this change is being made, can you add some description to the PR itself, so I can understand why this huge change is necessary?

anandjakhaniya commented:
> Since the Jira ticket has no reason as to why this change is being made, can you add some description to the PR itself, so I can understand why this huge change is necessary?

Added the following comment in the description.

> We have an artifact store with two different options which are S3 and NFS. so for that, we have created two services using spring boot to serve artifact store functionality. In the current situation, This functionality is used by two backends which are Registry and ModelDB and both the backends have their own code so whenever we have updated something in one place we have to manage it on the other backends as well. so to protect duplication and uniqueness of the code we need these changes in a common place so that both backends are used same code and we can manage the code maintainability from one place.

jkwatson-verta reviewed (approved):
When this is merged, will it break any of the other service repositories?

anandjakhaniya commented:
> When this is merged, will it break any of the other service repositories?

Yes it will break the registry backend. We have one registry PR related to this.

Commit:
Update protos from https://github.com/VertaAI/protos-all/pull/336

anandjakhaniya commented:
Open new PR with the latest main code where protos changes are not required as per @conradoverta suggestion  https://vertaai.slack.com/archives/C0306V04EJC/p1661487421380299?thread_ts=1661422409.689719&cid=C0306V04EJC 

New PR: https://github.com/VertaAI/modeldb/pull/3153
